indexer/src/main/scala/com/expedia/www/haystack/trace/indexer/writers/es/ElasticSearchWriter.scala

```diff
@@ -46,10 +47,11 @@ object ElasticSearchWriterUtils {
   // creates an index name based on current date. following example illustrates the naming convention of
   // elastic search indices:
   // haystack-span-2017-08-30-1
-  def indexName(prefix: String, indexHourBucket: Int): String = {
-    val currentTime = Calendar.getInstance(timezone)
-    val bucket: Int = currentTime.get(Calendar.HOUR_OF_DAY) / indexHourBucket
-    s"$prefix-${format.format(currentTime.getTime)}-$bucket"
+  def indexName(prefix: String, indexHourBucket: Int, eventTimeMicros: Long): String = {
+    val eventTimeCal = Calendar.getInstance(timezone)
```

Comment:
Can we use Joda DateTime instead of Calendar.getInstance() for better performance?

Reply:
done